pkg/ddc/alluxio/health_check.go

```diff
@@ -151,8 +155,14 @@ func (e *AlluxioEngine) checkMasterHealthy() (err error) {
 // checkWorkersHealthy check workers number changed
 func (e *AlluxioEngine) checkWorkersHealthy() (err error) {
 	// Check the status of workers
-	workers, err := kubeclient.GetStatefulSet(e.Client, e.getWorkerName(), e.namespace)
+	workers, err := ctrl.GetWorkersAsStatefulset(e.Client,
+		types.NamespacedName{Namespace: e.namespace, Name: e.getWorkerName()})
 	if err != nil {
+		if fluiderrs.IsDeprecated(err) {
+			e.Log.Info("Warning: Deprecated mode is not support, so skip handling", "details", err)
+			e.Recorder.Event(e.runtime, corev1.EventTypeWarning, common.RuntimeDeprecated, "Detected deprecated runtime, the status might not be up-to-date")
```

Comment:
`Detected deprecated runtime, the status might not be up-to-date` to `"The runtime is created by controllers before v0.7.0, to fully enable latest capabilities, please delete the runtime and create a new one "`